PKHUD/PKHUD.swift

```diff
@@ -144,15 +181,12 @@ open class PKHUD: NSObject {
         hide(animated: true, completion: completion)
     }
 
-    internal func startAnimatingContentView() {
-        if let animatingContentView = contentView as? PKHUDAnimating, isVisible {
-            animatingContentView.startAnimation()
-        }
-    }
+    internal func handleGraceTimer(_ timer: Timer? = nil) {
+        // Show the HUD only if the task is still running
 
-    internal func stopAnimatingContentView() {
-        if let animatingContentView = contentView as? PKHUDAnimating {
-            animatingContentView.stopAnimation?()
+        if !finished {
+            container.showFrameView()
```

Comment:
No. I think is better to use  [NSTimer.isValid](https://developer.apple.com/reference/foundation/timer/1408249-isvalid)